src/hooks/logging.rs

```diff
@@ -0,0 +1,45 @@
+use crate::{EvaluationContext, EvaluationDetails, EvaluationError, Value};
+
+use super::{Hook, HookContext, HookHints};
+
+/// A hook that logs the evaluation lifecycle of a flag.
+pub struct LoggingHook;
+
+#[async_trait::async_trait]
+impl Hook for LoggingHook {
+    async fn before<'a>(
+        &self,
+        context: &HookContext<'a>,
+        _: Option<&'a HookHints>,
+    ) -> Result<Option<EvaluationContext>, EvaluationError> {
+        log::debug!("Before hook for flag {}", context.flag_key);
```

Comment:
just thinking, wouldn't it be better to raise the log levels, `debug` -> `info` and the `trace` -> `debug`?

Reply:
It's debatable, enterprise clients usually hide `debug` level of logs in production. Flag evaluating could generate extremely many logs in large applications so it may force a client to disable logs specifically for the OpenFeature library because just setting log level `info` is not enough.
But, I don't insist :)